network/packet_listener_proxy.go

```diff
@@ -56,12 +56,28 @@ type packetListenerRequestSender struct {
 // [transport.PacketListener] and would like to inject it into one of the network stacks (for example,
 // network/lwip2transport) as UDP traffic handlers.
 func NewPacketProxyFromPacketListener(pl transport.PacketListener) (PacketProxy, error) {
+	return NewPacketProxyFromPacketListenerWithTimeout(pl, 30*time.Second)
+}
+
+// NewPacketProxyFromPacketListenerWithTimeout creates a new [PacketProxy] that uses the existing
+// [transport.PacketListener] to create connections to a proxy. You can also specify a timeout for UDP sessions created
+// by NewSession.
+//
+// This function is useful if you already have an implementation of [transport.PacketListener] and you want to use it
+// with one of the network stacks (for example, network/lwip2transport) as a UDP traffic handler.
+//
+// The timeout is the write idle timeout. This means that if there are no WriteTo operations on the UDP session created
+// by NewSession for the specified amount of time, the proxy will end this session.
+func NewPacketProxyFromPacketListenerWithTimeout(pl transport.PacketListener, timeout time.Duration) (PacketProxy, error) {
```

Comment:
This approach is not very scalable. What if there are other parameters in the future?
Since it's not a required parameter, let's provide a setter instead, like we do for the shadowsocks prefix.
Require params -> constructor, optional params -> setters.

There's the option of exposing the field, but setters are more flexible, as it allows us to migrate the implementation in the future if needed.

You will need to expose the concrete type in the return of `NewPacketProxyFromPacketListener()`.
Perhaps call it `ListenerPacketProxy` and have no exposed fields? Then one can call `ListenerPacketProxy.SetIdleTimeout(timeout)`.

Another option is to use the "Options" pattern, which is popular in Go, but it's more complicated. See
[Functional Options Pattern](https://golang.cafe/blog/golang-functional-options-pattern.html#:~:text=Option%203%3A%20Functional%20Options%20Pattern) for an explanation.

Reply:
Another thing to point out is that the expiration logic could be implemented as a wrapper around the PacketListener instead, which would allow for a more composable design and to be used in other contexts.

The TimeoutPacketListener would create a timer for each connection that will close the connection when idle. The consumer should get a ErrClosed when that happens and react accordingly.
The connections would be wrapped in order to reset the timer on writes.

That would make packet connections a lot more like stream conns.

Up to you, perhaps something for another PR in the future.

Reply:
Added, now we can use the following code to create a new packet proxy with a specific timeout. The option name `WithPacketListenerWriteIdleTimeout` is a little bit verbose because it is under `network` package.

```go
NewPacketProxyFromPacketListenerWithOptions(pl, WithPacketListenerWriteIdleTimeout(5 * time.Minute))
```